### PR TITLE
UIINREACH-19: INN-Reach Record Contribution Settings: Create Contribution Criteria Settings Form

### PR DESCRIPTION
**Pull request by Dmytro-Melnyshyn**
- added the functionality for creating/editing the form of contribution criteria settings
- added tests

### CODE REVIEW

File: src/settings/components/ContributionCriteria/ContributionCriteriaForm/ContributionCriteriaForm.js [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+  folioLocations,
+  statisticalCodes,
+  statisticalCodeTypes,
+  values,
+  initialValues,
+  isResetForm,
+  form,
+  serverSelection,
+  handleSubmit,
+  onAppointIsResetForm,
```
Comment (Godlevskyi) [inline src/settings/components/ContributionCriteria/ContributionCriteriaForm/ContributionCriteriaForm.js:39]: Please consider renaming onAppointIsResetForm and onAppointIsPristine to more meaningful names

File: src/settings/routes/ContributionCriteriaRoute/ContributionCriteriaCreateEditRoute.js [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+  const [initialValues, setInitialValues] = useState(DEFAULT_VALUES);
+  const [openModal, setOpenModal] = useState(false);
+  const [nextLocation, setNextLocation] = useState(null);
+  const [isResetForm, setIsResetForm] = useState(false);
+
+  const folioLocations = locations[0]?.locations || [];
+  const statisticalCodeTypes = statisticalCodeTypesData[0]?.statisticalCodeTypes || [];
+  const statisticalCodes = statisticalCodesData[0]?.statisticalCodes || [];
+
+  const appointIsResetForm = (value) => {
```
Comment (Godlevskyi) [inline src/settings/routes/ContributionCriteriaRoute/ContributionCriteriaCreateEditRoute.js:69]: Please change to to changeFormResetState

File: src/settings/routes/ContributionCriteriaRoute/ContributionCriteriaSelectionRoute.js [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+        path={`${match.path}/:id`}
+        render={props => (
+          <ContributionCriteriaCreateEditRoute
+            {...props}
+            isPristine={isPristine}
+            prevServerName={prevServerName}
+            centralServersOptions={centralServersOptions}
+            serverSelection={serverSelection}
+            onChangePrevServerName={changePrevServerName}
+            onFooter={getFooter}
```
Comment (Godlevskyi) [inline src/settings/routes/ContributionCriteriaRoute/ContributionCriteriaSelectionRoute.js:135]: Please change onFooter prop to ferderFooter as it is not a callback

File: src/settings/components/ContributionCriteria/ContributionCriteriaForm/ContributionCriteriaForm.js (hunk 1 above)
Comment (Dmytro-Melnyshyn) [reply src/settings/components/ContributionCriteria/ContributionCriteriaForm/ContributionCriteriaForm.js]: Thanks, fixed

File: src/settings/routes/ContributionCriteriaRoute/index.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+import ReactRouterPropTypes from 'react-router-prop-types';
+import { hot } from 'react-hot-loader';
+
+import {
+  Route,
+  Switch,
+} from '@folio/stripes/core';
+
+import ContributionCriteriaSelectionRoute from './ContributionCriteriaSelectionRoute';
+
```
[the remark this answers is not in the document]
Comment (Dmytro-Melnyshyn) [reply src/settings/routes/ContributionCriteriaRoute/index.js:11]: Thanks, fixed

File: src/settings/routes/ContributionCriteriaRoute/ContributionCriteriaCreateEditRoute.js (hunk 2 above)
Comment (Dmytro-Melnyshyn) [reply src/settings/routes/ContributionCriteriaRoute/ContributionCriteriaCreateEditRoute.js]: Thanks, fixed

File: src/settings/routes/ContributionCriteriaRoute/ContributionCriteriaSelectionRoute.js (hunk 3 above)
Comment (Dmytro-Melnyshyn) [reply src/settings/routes/ContributionCriteriaRoute/ContributionCriteriaSelectionRoute.js]: Thanks, fixed

File: src/settings/routes/ContributionCriteriaRoute/ContributionCriteriaCreateEditRoute.js [hunk 5]
```diff
@@ -0,0 +1,233 @@
...
+  };
+
+  useEffect(() => {
+    if (selectedServer.id) {
+      mutator.selectedServerId.replace(selectedServer.id);
+      setIsContributionCriteriaPending(true);
+
+      mutator.contributionCriteria.GET()
+        .then(response => setContributionCriteria(response))
+        .catch(() => {})
```
Comment (Godlevskyi) [inline src/settings/routes/ContributionCriteriaRoute/ContributionCriteriaCreateEditRoute.js:113]: Please add callout with an error here

**Review by Godlevskyi — APPROVED** [review]

**Pull request merged**